server/api/src/rsServerReport.cpp

```diff
@@ -456,34 +358,23 @@ irods::error get_resource_array(
         }
 
 
-        json_t* entry = json_object();
-        if ( !entry ) {
-            return ERROR(
-                       SYS_MALLOC_ERR,
-                       "failed to alloc entry" );
-        }
+        auto entry = json::object();
 
         ret = serialize_resource_plugin_to_json(itr->second, entry);
         if(!ret.ok()) {
             ret = PASS(ret);
             irods::log(ret);
-            json_object_set_new(entry, "ERROR", json_string(ret.result().c_str()));
+            entry["ERROR"] = ret.result();
         }
 
-        json_array_append_new( _resources, entry );
-
+        _resources.push_back(entry);
     } // for itr
 
     return SUCCESS();
-
 } // get_resource_array
 
-
-
-irods::error get_file_contents(
-    const std::string _fn,
-    std::string&       _cont ) {
-
+irods::error get_file_contents( const std::string _fn, std::string& _cont )
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter '_fn' should be passed by const reference.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4751251)